core/lib/dal/src/blocks_dal.rs

```diff
@@ -756,6 +920,52 @@ impl BlocksDal<'_, '_> {
         Ok(())
     }
 
+    pub async fn get_ready_for_precommit_txs(
+        &mut self,
+        l1_batch_number: Option<L1BatchNumber>,
+    ) -> DalResult<Vec<TxForPrecommit>> {
+        let mut tx = self.storage.start_transaction().await?;
+        // Miniblocks belongs to the non sealed batches don't have batch number,
+        // so for the pending batch we use None
+
+        let txs = sqlx::query!(
+            r#"
+            SELECT
+                transactions.hash, transactions.error,
+                miniblock_number AS "miniblock_number!",
+                miniblocks.timestamp
+            FROM transactions
+            JOIN miniblocks ON transactions.miniblock_number = miniblocks.number
+            WHERE
+                (
+                    transactions.l1_batch_number IS NULL AND $2
+                    OR transactions.l1_batch_number = $1
```

Comment:
there is no index on `transactions.l1_batch_number` so this query will do sequential scan

Reply:
I didn't mean to add an index, it will affect sequncing performance; it's possible to do it without an index (e.g. join with miniblocks and use miniblocks.l1_batch_number)

Reply:
good idea